packages/driver/src/cypress/cy.ts

```diff
@@ -550,6 +550,22 @@ export class $Cy extends EventEmitter2 implements ITimeouts, IStability, IAssert
 
         setWindowDocumentProps(autWindow, this.state)
 
+        if (this.Cypress.isBrowser('webkit')) {
+          // WebKit's unhandledrejection event will sometimes not fire within the AUT
+          // due to a documented bug: https://bugs.webkit.org/show_bug.cgi?id=187822
+          // To ensure that the event will always fire (and always report these
+          // unhandled rejections to the user), we patch the AUT's Error constructor
+          // to enqueue a no-op microtask when executed, which ensures that the unhandledrejection
+          // event handler will be executed if this Error is uncaught.
+          let originalError = autWindow.Error
+
+          autWindow.Error = function (...args) {
+            autWindow.queueMicrotask(() => {})
```

Comment:
Looking at this now, I wondered if this could backfire and cause an infinitely expanding queue, but I can't think of a scenario where that would happen where the user wouldn't already be doing infinite recursion anyways. So this seems safe enough...

Reply:
Ah, I thought of one - if `window.queueMicrotask` is monkey-patched to throw an Error itself, this would infinitely recurse. But that seems unlikely to happen in the wild. Maybe we could name this function something like `__CyWebKitError` so that it's at least identifiable in a stack trace in case someone does run in to this?

Reply:
That does seem unlikely, but naming the function is a good idea.

Reply:
Could even `try...catch` the `queueMicrotask` call... but... then that could also lead to really hard-to-debug issues ("why does my queueMicrotask monkey-patch make unhandled rejections not get caught by Cypress???"), so I think the way you have it is good for now. Let's see if any bug reports roll in.